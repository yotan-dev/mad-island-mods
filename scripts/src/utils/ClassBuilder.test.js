import { describe, it, expect } from '@jest/globals';
import { ClassBuilder } from './ClassBuilder.js';
import { FieldBuilder } from './FieldBuilder.js';

describe('ClassBuilder', () => {
	it('should build class with namespace, class name and int const fields', () => {
		const cb = new ClassBuilder('YotanModCore.Consts', 'SampleClass');
		cb.addIntConstField('Test', 0, 'This is a test');
		cb.addIntConstField('Test2', 1);
		expect(cb.build()).toMatchSnapshot();
	});

	it('should build class with namespace, class name and int const fields with attributes', () => {
		const cb = new ClassBuilder('YotanModCore.Consts', 'SampleClass');
		cb.addRawField(FieldBuilder.intConst('Test', 0).addAttribute('[Obsolete]').build());
		expect(cb.build()).toMatchSnapshot();
	});
});
