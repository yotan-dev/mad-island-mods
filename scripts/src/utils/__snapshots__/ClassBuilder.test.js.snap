// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ClassBuilder should build class with namespace, class name and int const fields 1`] = `
"// Automatically generated - Do NOT edit.
namespace YotanModCore.Consts
{
	public static class SampleClass
	{
		public const int Test = 0; // This is a test

		public const int Test2 = 1;
	}
}
"
`;

exports[`ClassBuilder should build class with namespace, class name and int const fields with attributes 1`] = `
"// Automatically generated - Do NOT edit.
namespace YotanModCore.Consts
{
	public static class SampleClass
	{
		[Obsolete]
		public const int Test = 0;
	}
}
"
`;
